barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_format.cpp

```diff
@@ -642,12 +581,23 @@ template <> UltraCircuitBuilder create_circuit(AcirProgram& program, const Progr
     BB_BENCH();
     AcirFormat& constraints = program.constraints;
     WitnessVector& witness = program.witness;
+    const bool is_write_vk_mode = witness.empty();
 
-    Builder builder{ metadata.size_hint, witness, constraints.public_inputs, constraints.varnum };
+    if (!is_write_vk_mode) {
+        BB_ASSERT_EQ(witness.size(),
+                     constraints.max_witness_index + 1,
+                     "ACIR witness size (" << witness.size() << ") does not match max witness index ("
+                                           << constraints.max_witness_index << ").");
+    } else {
+        witness.resize(constraints.max_witness_index + 1, 0);
```

Comment:
I wonder if there is some easy way around this. Seems a shame to have to initialize this potentially large vector just to make things not break. Not a high priority though